backend/tests/sessions/test_sessions_routes.py

```diff
@@ -426,3 +436,78 @@ def test_create_persistent_jupyter_session(client, db, user, kubernetes):
         kubernetes.sessions[0]["docker_image"]
         == "jupyter/minimal-notebook:python-3.10.8"
     )
+
+
+def test_group_models_by_tool_version():
+    first_tool = DatabaseVersion(id=1)
+    second_tool = DatabaseVersion(id=2)
+
+    models = [
+        DatabaseCapellaModel(
+            name="first",
+            version=first_tool,
+        ),
+        DatabaseCapellaModel(
+            name="second-1",
+            version=second_tool,
+        ),
+        DatabaseCapellaModel(
+            name="second-2",
+            version=second_tool,
+        ),
+    ]
+
+    models_by_tool = sessions_routes.group_models_by_tool_version(models)
+
+    assert models_by_tool == [[models[0]], [models[1], models[2]]]
+
+
+def test_provision_sessions_as_user(
+    client: testclient.TestClient,
+    db,
+    user,
+    kubernetes,
+):
+    capella_version = next(
+        v
+        for v in get_versions(db)
+        if v.tool.name == "Capella" and v.name == "6.0.0"
+    )
+
+    jupyter_version = next(
+        v for v in get_versions(db) if v.tool.name == "Jupyter"
+    )
+
+    capella_model, capella_git_model = setup_git_model_for_user(
+        db, user, capella_version
+    )
+    jupyter_model, jupyter_git_model = setup_model(
+        db, capella_model.project, jupyter_version
+    )
+
+    response = client.post(
+        f"/api/v1/projects/{capella_model.project.slug}/sessions/provision",
+        json={
+            "models": [
+                {
+                    "model_slug": capella_model.slug,
+                },
+                {
+                    "model_slug": jupyter_model.slug,
+                },
+            ],
+            "persistent_workspace": True,
+        },
+    )
+    sessions = response.json()
+    print(sessions)
```

Comment:
```suggestion
```

There is an additional `print` in the `test_create_read_only_session_as_user` where it would be great if you could remove this as well.